packages/plans-grid-next/src/components/actions.tsx

```diff
@@ -454,59 +354,69 @@ const PlanFeaturesActionsButton: React.FC< PlanFeaturesActionsButtonProps > = (
 		storageAddOnsForPlan,
 	} );
 
-	const onCtaClick = useActionCallback( {
+	const priceString = formatCurrency(
+		( discountedPrice.monthly || originalPrice.monthly ) ?? 0,
+		currencyCode || 'USD',
+		{
+			stripZeros: true,
+			isSmallestUnit: true,
+		}
+	);
+
+	const { callback, text, postButtonText, status } = useAction( {
+		// TODO: Double check that we need to do this boolean coercion
+		availableForPurchase,
+		isLargeCurrency: !! isLargeCurrency,
+		isStuck,
 		planSlug,
+		planTitle,
+		priceString,
 		cartItemForPlan,
 		selectedStorageAddOn,
 	} );
 
-	// TODO: Unsure about using free plan as a fallback. We should revisit.
-	const onFreeTrialCtaClick = useActionCallback( {
+	const { callback: freeTrialCallback, text: freeTrialText } = useAction( {
+		// TODO: Double check that we need to do this boolean coercion
+		isFreeTrialAction: true,
+		isLargeCurrency: !! isLargeCurrency,
+		isStuck,
+		// TODO: Unsure about using free plan as a fallback. We should revisit.
 		planSlug: freeTrialPlanSlug ?? PLAN_FREE,
+		planTitle,
+		priceString,
 		cartItemForPlan: { product_slug: freeTrialPlanSlug ?? PLAN_FREE },
+		selectedStorageAddOn,
 	} );
 
 	if ( isWpcomEnterpriseGridPlan( planSlug ) ) {
 		return (
-			<PlanButton planSlug={ planSlug } onClick={ onCtaClick }>
+			<PlanButton planSlug={ planSlug } onClick={ callback }>
 				{ translate( 'Learn more' ) }
```

Comment:
This can be the `text` from the `useAction` hook. So this condition certainly candidate for removing in the end.

Reply:
[46fe3b8](https://github.com/Automattic/wp-calypso/pull/89926/commits/46fe3b87bbc88aa8ea8fe0cf31e71908f0bc08de)

Reply:
[46fe3b8](https://github.com/Automattic/wp-calypso/pull/89926/commits/46fe3b87bbc88aa8ea8fe0cf31e71908f0bc08de)